cargo-dist/src/tasks.rs

```diff
@@ -2558,79 +2669,80 @@ pub(crate) fn parse_tag(
-            let info = &graph.workspace().package(some_pkg);
-            let some_tag = format!(
-                "--tag={}-v{}",
-                info.name,
-                info.version.as_ref().unwrap().cargo()
-            );
+/// Get a help printout for what --tags could have been passed
+fn tag_help(
+    graph: &DistGraphBuilder,
+    versions: SortedMap<&Version, Vec<PackageIdx>>,
+    base_suggestion: &str,
+) -> String {
+    use std::fmt::Write;
+    let mut help = String::new();
 
-            writeln!(
-                help,
-                "you can also request any single package with {some_tag}"
-            )
-            .unwrap();
-            return Err(DistError::TooManyUnrelatedApps { help });
-        } else {
-            // We don't need a coherent announcement tag so use a fake one to continue on
-            announcement_tag = Some("v1.0.0-FAKEVER".to_owned());
-            announcing_prerelease = true;
-            announcing_version = Some("1.0.0-FAKEVER".parse().unwrap());
+    let Some(some_pkg) = versions
+        .first_key_value()
+        .and_then(|(_, packages)| packages.first())
+    else {
+        return r#"It appears that you have no packages in your workspace with distable binaries. You can rerun with "--verbose=info" to see what cargo-dist thinks is in your workspace. Here are some typical issues:
+
+    If you're trying to use cargo-dist to announce libraries, we require you explicitly select the library with e.g. "--tag=my-library-v1.0.0", as this mode is experimental.
+
+    If you have binaries in your workspace, `publish = false` could be hiding them and adding "dist = true" to [package.metadata.dist] in your Cargo.toml may help."#.to_owned();
+    };
+
+    help.push_str(base_suggestion);
+    help.push_str("\n\n");
+    help.push_str("Here are some options:\n\n");
+    for (version, packages) in &versions {
+        write!(help, "--tag=v{version} will Announce: ").unwrap();
+        let mut multi_package = false;
+        for &pkg_id in packages {
+            let info = &graph.workspace().package(pkg_id);
+            if multi_package {
+                write!(help, ", ").unwrap();
+            } else {
+                multi_package = true;
+            }
+            write!(help, "{}", info.name).unwrap();
         }
+        writeln!(help).unwrap();
     }
-    Ok(AnnouncementTag {
-        tag: announcement_tag.expect("integrity error: failed to select announcement tag"),
-        prerelease: announcing_prerelease,
-        version: announcing_version,
-        package: announcing_package,
-        rust_releases,
-    })
+    help.push('\n');
+    let info = &graph.workspace().package(*some_pkg);
+    let some_tag = format!(
+        "--tag={}-v{}",
+        info.name,
+        info.version.as_ref().unwrap().cargo()
+    );
+
+    writeln!(
+        help,
+        "you can also request any single package with {some_tag}"
+    )
+    .unwrap();
```

Comment:
is there a reason to use writeln + unwrap over println? (same above)

Reply:
We need to shove the result into an Error type for miette.